src/MeetingsSDKAdapter.js

```diff
@@ -216,19 +216,26 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
    * @param {string} ID ID of the meeting to mute audio
    * @memberof MeetingsSDKAdapter
    */
-  muteAudioMeeting(ID) {
+  async handleLocalAudio(ID) {
     const sdkMeeting = this.fetchMeeting(ID);
+    const state = this.meetings[ID].localAudio.getAudioTracks()[0].enabled;
+    let eventName;
 
-    sdkMeeting
-      .muteAudio()
-      .then(() => {
-        // Due to SDK limitation around promises, we need to emit a custom event for audio mute action
-        sdkMeeting.emit(EVENT_MEDIA_LOCAL_MUTE, {control: MUTE_AUDIO_CONTROL});
-      })
-      .catch((error) => {
-        // eslint-disable-next-line no-console
-        console.error(`Unable to mute audio for meeting "${ID}"`, error);
-      });
+    try {
+      if (state) {
+        await sdkMeeting.muteAudio();
+        eventName = EVENT_MEDIA_LOCAL_MUTE;
+      } else {
+        await sdkMeeting.unmuteAudio();
+        eventName = EVENT_MEDIA_LOCAL_UNMUTE;
```

Comment:
Why do you need to have two separate events? You can distinguish from the data passed in what you need to do without listening/emitting for two separate events